src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -262,63 +344,103 @@ class npc_mograine : public CreatureScript
+                summonedMograine->StopMovingOnCurrentPos();
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
+                summonedMograine->AI()->Talk(SAY_HM_AB_TALK0, 214ms);
+                events.ScheduleEvent(EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE, 3024ms);
+                break;
+            case EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE:
+                me->SetFacingToObject(summonedMograine);
+                events.ScheduleEvent(EVENT_MOGRAINE_UNIT_STAND_STATE_STAND, 403ms);
+                break;
+            case EVENT_MOGRAINE_UNIT_STAND_STATE_STAND:
+                me->SetStandState(UNIT_STAND_STATE_STAND);
+                me->SetSheath(SHEATH_STATE_MELEE);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK4, 809ms);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK4:
+                me->HandleEmoteCommand(EMOTE_ONESHOT_QUESTION);
+                me->AI()->Talk(SAY_MO_AB_TALK4, 209ms);
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE_TALK, 4636ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_EMOTE_TALK:
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_QUESTION);
+                summonedMograine->AI()->Talk(SAY_HM_AB_TALK1, me, 10ms);
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE1, 3429ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_EMOTE1:
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_QUESTION);
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE2, 3236ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_EMOTE2:
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE3, 3243ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_EMOTE3:
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_ROAR);
+                events.ScheduleEvent(EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE2, 3232ms);
+                break;
+            case EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE2:
+                me->SetFacingToObject(summonedMograine);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK5, 1210ms);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK5:
+                me->SetSheath(SHEATH_STATE_UNARMED);
+                me->HandleEmoteCommand(EMOTE_ONESHOT_BEG);
+                me->AI()->Talk(SAY_MO_AB_TALK5, 214ms);
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_CASTSPELL, 3022ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_CASTSPELL:
+                summonedMograine->CastSpell(me, SPELL_FORGIVENESS, false);
+                //events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_KILL_MOGRAINE, 1010ms);
+                events.ScheduleEvent(EVENT_MOGRAINE_CASTSPELL, 1000ms); // case 16 hack fix time
```

Comment:
Why

Reply:
After SPELL_FORGIVENESS is released on Blizzard's server, Mograine has a visual effect of being hit by a lightning spell. Mograine dies after 1010ms. After using SPELL_FORGIVENESS in the script, mogranine does not have lightning spells. Combined with the visual effect of using this ability as a spell in the previous code, I added this spell. And did

Reply:
If in the future SPELL_FORGIVENESS after hitting the target, there is a lightning hit effect. You can jump to EVENT_HIGHLORD_MOGRAINE_KILL_MOGRAINE